src/Microsoft.CodeQuality.Analyzers.Exp/Core/Globalization/DoNotPassLiteralsAsLocalizedParameters.cs

```diff
@@ -0,0 +1,358 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftGlobalizationAnalyzersResources.DoNotPassLiteralsAsLocalizedParametersMessage), MicrosoftGlobalizationAnalyzersResources.ResourceManager, typeof(MicrosoftGlobalizationAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftGlobalizationAnalyzersResources.DoNotPassLiteralsAsLocalizedParametersDescription), MicrosoftGlobalizationAnalyzersResources.ResourceManager, typeof(MicrosoftGlobalizationAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Globalization,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: "https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1303-do-not-pass-literals-as-localized-parameters",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                INamedTypeSymbol localizableStateAttributeSymbol = WellKnownTypes.LocalizableAttribute(compilationContext.Compilation);
+                INamedTypeSymbol conditionalAttributeSymbol = WellKnownTypes.ConditionalAttribute(compilationContext.Compilation);
+                INamedTypeSymbol systemConsoleSymbol = WellKnownTypes.Console(compilationContext.Compilation);
+                ImmutableHashSet<INamedTypeSymbol> typesToIgnore = GetTypesToIgnore(compilationContext.Compilation);
+
+                compilationContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    if (!(operationBlockStartContext.OwningSymbol is IMethodSymbol containingMethod))
```

Comment:
Is this FxCop compat? This doesn't seem like it's correct, we create LocalizableResourceStrings as fields in this very class.

Reply:
Yes, this is not FxCop compat, but restriction of our initial flow analysis implementation that doesn't work on initializers. Let me file a tracking issue to remove this workaround.